modules/apps/ccv/utils/utils.go

```diff
@@ -0,0 +1,24 @@
+package utils
+
+import (
+	abci "github.com/tendermint/tendermint/abci/types"
+)
+
+func AccumulateChanges(currentChanges, newChanges []abci.ValidatorUpdate) []abci.ValidatorUpdate {
+	m := make(map[string]abci.ValidatorUpdate)
+
+	for i := 0; i < len(currentChanges); i++ {
+		m[currentChanges[i].PubKey.String()] = currentChanges[i]
+	}
+
+	for i := 0; i < len(newChanges); i++ {
+		m[newChanges[i].PubKey.String()] = newChanges[i]
+	}
+
+	var out []abci.ValidatorUpdate
+
+	for _, update := range m {
```

Comment:
Is this used in the state machine? Iteration over maps is nondeterministic, seems likely this could result in app hash mismatch